### PR TITLE
[LUA]Move BST JA to jobUtils file.

### PR DESCRIPTION
**Pull request by hooksta4**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
- Creates Beastmaster jobUtils
- Moves all BST job abilities into jobUtils
- Removes all logic from individual job abilities and puts them in jobUtils
- Capture shows beast under sync are not able to call jug pet greater than sync'd level (currently allowed to do so) with @Xaver-DaRed 
    - two beastmasters sync'd with pets, used JA leave and tried to call 28 level jub while sync'd at 23. Retail returned message CANT_BE_USED_IN_AREA
 

## Steps to test these changes
Moves all job ability logic into centralized location (jobUtils)

### CODE REVIEW

File: scripts/actions/abilities/gauge.lua [hunk 1]
```diff
@@ -8,27 +8,11 @@
 local abilityObject = {}
 
 abilityObject.onAbilityCheck = function(player, target, ability)
-    if player:getPet() ~= nil then
-        return xi.msg.basic.ALREADY_HAS_A_PET, 0
-    end
-
-    return 0, 0
+    return xi.job_utils.beastmaster.onAbilityCheckGuage(player, target, ability)
```
Comment (WinterSolstice8) [inline scripts/actions/abilities/gauge.lua]: Gauge is typoed here and in the job_utils lua

File: scripts/globals/job_utils/beastmaster.lua [hunk 2]
```diff
@@ -0,0 +1,512 @@
...
+    return 0, 0
+end
+
+-- On Ability Use Unleash
+xi.job_utils.beastmaster.onUseAbilityUnleash = function(player, target, ability)
+    player:addStatusEffect(xi.effect.UNLEASH, 9, 0, 60)
+end
+
+-- On Ability Check For Leave, Heel and Stay.
+xi.job_utils.beastmaster.onAbilityCheckNilPet = function(player, target, ability)
```
Comment (WinterSolstice8) [inline scripts/globals/job_utils/beastmaster.lua:420]: This function needs to check if your pet is a jug pet (or charmed pet). Leave/Heel/Stay shouldn't work on drg wyverns, automatons, summoner avatars, etc.

Comment (WinterSolstice8) [reply scripts/globals/job_utils/beastmaster.lua:420]: It looks like other functions need that check too, might need a small refactor.

File: scripts/actions/abilities/gauge.lua (hunk 1 above)
Comment (hooksta4) [reply scripts/actions/abilities/gauge.lua]: least i was consistent ;)

Comment (WinterSolstice8) [reply scripts/actions/abilities/gauge.lua]: Yup haha

File: scripts/globals/job_utils/beastmaster.lua (hunk 2 above)
Comment (hooksta4) [reply scripts/globals/job_utils/beastmaster.lua:420]: I think I understand what you are saying, for this PR, I didn't change anything that was already in current code base *except* for the level sync change. Not sure if that is in the scope of this PR?

Comment (hooksta4) [reply scripts/globals/job_utils/beastmaster.lua:420]: BUT, would it be something like this?

`        if player:isJugPet() and
        pet:getObjType() ~= xi.objType.MOB`

Comment (hooksta4) [reply scripts/globals/job_utils/beastmaster.lua:420]: although arent all the commands different for each like leave/release/dismiss/deactivate all from main Job?

Comment (WinterSolstice8) [reply scripts/globals/job_utils/beastmaster.lua:420]: For generic ability check for "can use ability"
```
if
    player:isJugPet() or
    pet:getObjType() == xi.objType.MOB
then
```
I think the one you pasted is from familiar, which does behave differently for jug vs non-jug(?)

Comment (WinterSolstice8) [reply scripts/globals/job_utils/beastmaster.lua:420]: p.s. I hate "isJugPet", but thats not our problem here :)

Comment (WinterSolstice8) [reply scripts/globals/job_utils/beastmaster.lua:420]: > although arent all the commands different for each like leave/release/dismiss/deactivate all from main Job?

They are, but for some reason there's conditions where they show up when they should not. I recently saw Dismiss as DRG/PUP or PUP/DRG with an automation out. It's just a safety thing until I can find time to fix that

Comment (hooksta4) [reply scripts/globals/job_utils/beastmaster.lua:420]: Interesting - i just went on a bst/pup and didnt see it in menu - not saying its not there but

Comment (WinterSolstice8) [reply scripts/globals/job_utils/beastmaster.lua:420]: It's pure paranoia, not gonna lie
absolutely not me when I see a SMN/BST

Comment (hooksta4) [reply scripts/globals/job_utils/beastmaster.lua:420]: haha love it!

**Comment by hooksta4** [comment]
Quick update: Made a TODO list in On Ability Check Reward: It is not returning the correct message.

**Review by WinterSolstice8 — APPROVED** [review]

**Pull request merged**